src/openapi-generator.ts

```diff
@@ -54,11 +53,15 @@ import { enumInfo } from './lib/enum-info';
 // See https://github.com/colinhacks/zod/blob/9eb7eb136f3e702e86f030e6984ef20d4d8521b6/src/types.ts#L1370
 type UnknownKeysParam = 'passthrough' | 'strict' | 'strip';
 
+const openApiVersions = ['3.0.0', '3.0.1', '3.0.2', '3.0.3', '3.1.0'] as const;
+
+export type OpenApiVersion = typeof openApiVersions[number];
+
 // This is essentially OpenAPIObject without the components and paths keys.
 // Omit does not work, since OpenAPIObject extends ISpecificationExtension
 // and is inferred as { [key: number]: any; [key: string]: any }
-interface OpenAPIObjectConfig {
-  openapi: string;
+export interface OpenAPIObjectConfig {
```

Comment:
Hmmm now that you updated the library I believe we can use Omit here. Because the ISpecificationExtension now has a key type of `x-${string}`. Can you check.

Reply:
Ahhhhh string template types. Should've thought of that 🧠